src/bun.js/bindings/sqlite/JSSQLStatement.cpp

```diff
@@ -452,13 +459,28 @@ static void initializeColumnNames(JSC::JSGlobalObject* lexicalGlobalObject, JSSQ
                 break;
             }
 
-            columnNames->add(Identifier::fromString(vm, WTF::String::fromUTF8({ name, len })));
+            // When joining multiple tables, the same column names can appear multiple times
+            // columnNames de-dupes property names internally
+            // We can't have two properties with the same name, so we use columnOffsets to track this.
+            int preCount = columnNames->size();
+            columnNames->add(
+                Identifier::fromString(vm, WTF::String::fromUTF8({name, len}))
+            );
+            int curCount = columnNames->size();
```

Comment:
```suggestion
            auto curCount = columnNames->size();
```

its actually `size_t`